### DISCUSSION TITLE
【任务】11-12月社区激励任务｜50,000 $PEOPLE

### DISCUSSION THREAD

**Discussion by AmazingAng** [AmazingAng/WTF-Solidity#301] [General]
从11月开始，WTF学院每月的社区激励上调到25,000 $PEOPLE，人物包括审核，改善WTF极简教程，也包括日常的PR/issue。这部分奖励优先分配给月度社区激励任务，剩余分给当月贡献者。

任务1：共同创作 WTF Solidity 合约安全篇

目前我们[WTF Solidity 合约安全篇](https://github.com/AmazingAng/WTF-Solidity)已经更新了6讲，按照教程计划，会更新约20个常见漏洞。这个任务是希望WTF贡献者协助我 [0xAA] 一起创作。

**要求**：
1. WTF成员 [填表](https://docs.google.com/forms/d/e/1FAIpQLSe4KGT8Sh6sJ7hedQRuIYirOoZK_85miz3dw7vA1-YjodgJ-A/viewform)
2. 完成 WTF Solidity 入门和进阶考试 [网站](https://wtf.academy)
3. WTF 贡献者 （参与过社区任务或完成过PR），或有智能合约安全相关工作经验。

**赏金**：15,000 $PEOPLE 代币，根据工作量分配，每人最多同时认领2讲。

认领情况：
伪随机数漏洞
@AmazingAng 

绕过合约长度检验
@AmazingAng 

DoS攻击
[hubingliang](https://github.com/hubingliang)

front-running
[hubingliang](https://github.com/hubingliang)

Silent Failing Sends and Unchecked-Send attacks
[LIPUU](https://github.com/LIPUU)

操纵block timestamp（Time Manipulation ）
[cheng521521](https://github.com/cheng521521)

短地址攻击（Short Address Attack）
[0x3f97](https://github.com/0x3f97)

预言机攻击
[chaochao214](https://github.com/chaochao214)

tx.origin钓鱼攻击
[seasidejuvenile666](https://github.com/seasidejuvenile666)

貔貅（honey pot）
[seasidejuvenile666](https://github.com/seasidejuvenile666)

任务2：共同创作 WTF Javascript 教程 入门篇

**要求**：
1. WTF成员 [填表](https://docs.google.com/forms/d/e/1FAIpQLSe4KGT8Sh6sJ7hedQRuIYirOoZK_85miz3dw7vA1-YjodgJ-A/viewform)
2. 有前端经验，写过教程/总结文章，或参加过WTF HTML/CSS教程创作。
3. 需要积极参与相关讨论。

**赏金**：20,000 $PEOPLE 代币，根据工作量分配，每人最多同时认领2讲。

1. Hello Javascript!  [lesenelir](https://github.com/lesenelir)
2. 变量声明 [seasidejuvenile666](https://github.com/seasidejuvenile666)
3.  常用类型 [0xb00lean](https://github.com/0xb00lean)
4. 运算符 [0xb00lean](https://github.com/0xb00lean)
5. 函数 [seasidejuvenile666](https://github.com/seasidejuvenile666)
6. 数组 [shengxinjing](https://github.com/shengxinjing)
7. 对象 [shengxinjing](https://github.com/shengxinjing)
8. 控制流 [holazz](https://github.com/holazz)
9. 异步 Promise async await [holazz](https://github.com/holazz)
10. node入门 [cheng521521](https://github.com/cheng521521)
11. 实战1: 铸造NFT脚本 @AmazingAng

## 认领方式
在这个帖子下面回复就能认领。

有用的资料：
1. Markdown书写规则（**必看**）：https://www.markdownguide.org/cheat-sheet/
2. 审核并添加图片后的例子：https://github.com/AmazingAng/WTFSolidity/tree/main/02_ValueTypes
3. 审核第一期：https://github.com/AmazingAng/WTFSolidity/discussions/18
4. https://github.com/AmazingAng/WTFSolidity/discussions/132
5. 8月社区激励任务：https://github.com/AmazingAng/WTF-Solidity/discussions/174

奖励发放:
这个月的两个奖励已在WTF Academy回溯性资助中分配: https://github.com/AmazingAng/WTF-Solidity/discussions/482

**Comment by hubingliang** [comment]
DoS攻击 front-running

**Reply by AmazingAng** [reply]
好的

**Comment by seasidejuvenile666** [comment]
认领tx.origin钓鱼攻击和貔貅（honey pot）

**Comment by cheng521521** [comment]
认领 操纵block timestamp（Time Manipulation ）

**Comment by twcctop** [comment]
认领 预言机攻击

**Comment by Camusugar** [comment]
Silent Failing Sends and Unchecked-Send attacks 认领

**Comment by 0x3f97** [comment]
认领 短地址攻击

**Comment by SimonsJ4r** [comment]
请问想认领的话从哪个列表中去认领？

**Reply by AmazingAng** [reply]
就在这个帖子中回复认领 不过目前任务被领完了 等下一个吧

**Comment by 0xb00lean** [comment]
认领 “基本数据类型”

**Comment by seasidejuvenile666** [comment]
认领
2.变量和常量
4.函数

**Comment by shengxinjing** [comment]
认领
5. 数组和集合
6. 对象和映射

**Comment by holazz** [comment]
认领
7. 控制流
8. 异步 Promise async await

**Comment by lesenelir** [comment]
认领
1. Hello Javascript!

**Comment by cheng521521** [comment]
认领9

**Reply by cheng521521** [reply]
认领9node入门
